custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/IndentationCheck.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.tools.analysis.checkstyle;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Hashtable;
+import java.util.Map;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openhab.tools.analysis.checkstyle.api.AbstractStaticCheck;
+
+import com.google.common.collect.ImmutableList;
+import com.puppycrawl.tools.checkstyle.api.FileText;
+
+/**
+ * Checks whether whitespace characters are use instead of tabs in xml files
+ * indentations and generates warnings in such cases.
+ *
+ * @author Lyubomir Papazov - initial contribution
+ * @author Kristina Simova - Removed REGEX and changed the way we look for indentation
+ * @author Velin Yordanov - Made the check more generic and added exceptions property
+ *
+ */
+public class IndentationCheck extends AbstractStaticCheck {
+    private static final String TAB_CHARACTER = "\t";
+    private static final String SPACE_CHARACTER = " ";
+    private static final String TABS = "tabs";
+    private static final String WARNING_MESSAGE = "Not supported indentation characters used";
+    private boolean onlyShowFirstWarning;
+    private Map<String, String> fileExtensionsToIndentation =  new Hashtable<String, String>();
```

Comment:
Please use the diamond operator, ie replace the second occurrence of the same generic by `<>`

Reply:
There is a [DiamondOperatorForVariableDefinitionCheck](https://sevntu-checkstyle.github.io/sevntu.checkstyle/apidocs/com/github/sevntu/checkstyle/checks/coding/DiamondOperatorForVariableDefinitionCheck.html) for this in [Sevntu-Checkstyle](http://sevntu-checkstyle.github.io/sevntu.checkstyle/). :smile:  
Did anyone ever have a look at and consider using any of those checks?